src/groups/semidirect_product_group.jl

```diff
@@ -107,14 +111,15 @@ function translate_diff!(G::SemidirectProductGroup, vout, x, y, v, conv::LeftAct
     N = submanifold(PG, 1)
     H = submanifold(PG, 2)
     A = G.op.action
-    nx, hx = submanifold_component.(Ref(x), (1, 2))
-    ny, hy = submanifold_component.(Ref(y), (1, 2))
-    nv, hv = submanifold_component.(Ref(v), (1, 2))
-    nvout, hvout = submanifold_component.(Ref(vout), (1, 2))
+    nx, hx = _subarray.(Ref(G), Ref(x), Val.((1, 2)))
```

Comment:
I think a simple `ntuple(i -> _subarray(G, x, Val(i)), Val(2))` would be cleaner here